Added UI elements

AtlantiaKing opened the pull request:
Added Canvas and CanvasActions as parent UI components
Added RectTransform, Image and Button as UI elements

AtlantiaKing committed:
Added canvas and canvas elements

Added canvas with width/height matching (it keeps the reference resolution)
Added CanvasElement to set the sprite size using the reference of the canvas
Currently only works with one child under the canvas

AtlantiaKing committed:
Added RectTransform

Canvas scaling now works for a full children tree of RectTransforms instead of only children of the canvas
Fixed GetComponent in GameObject not working
Renamed SpriteRendererComponent to Image
Added RectTransform component
Deleted CanvasElement component
Added Stretch MatchMode in the CanvasComponent

AtlantiaKing committed:
Merge branch 'main' into ui-elements

AtlantiaKing committed:
Added clickable button

Added CanvasActions component which handles all input for the whole canvas
Added ICanvasElement interface which presents a clickable object, these kind of objects will be saved inside the CanvasActions component
Added Button component which derives from ICanvasElement, it adds itself to the CanvasActions component

AtlantiaKing committed:
Changed priority of button clicks

Element click now still happens even though the cursor is outside the element

LeeVangraefschepe commented on LeapEngine/Components/RenderComponents/UIComponents/Button.h:
```diff
@@ -0,0 +1,57 @@
+		void SetHoverColor(const glm::vec4& color);
+		void SetClickColor(const glm::vec4& color);
+
+		void SetVisuals(Image* pVisual);
```
Visuals weird name to call image an image?

LeeVangraefschepe reviewed (approved):
Well done all the headers are clean (barely any includes). Love the event usage for the rect transform that checks for the window rescale and overal canvas. Check the comment for a variable rename suggestion overal well done!

AtlantiaKing committed:
Renamed Visual to Image in Button

Renamed m_pVisual button inside the Button component class to m_pImage
Renamed SetVisuals method to SetImage inside the Button component class

AtlantiaKing committed:
Fixed build error
